src/tools/rbd_mirror/image_sync/ImageCopyRequest.cc

```diff
@@ -146,7 +148,20 @@ void ImageCopyRequest<I>::send_object_copies() {
       }
     }
     complete = (m_current_ops == 0);
+
+    if (!complete) {
+      m_update_sync_ctx = new FunctionContext([this](int r) {
+          this->send_update_sync_point();
+      });
+    }
+  }
+
+  if (m_update_sync_ctx) {
+    Mutex::Locker l(*m_timer_lock);
+    m_timer->add_event_after(m_update_sync_point_interval,
```

Comment:
I don't think I need to hold both locks, although I need to acquire `m_timer_lock` before testing `m_update_sync_ctx != nullptr`.
If the sync completes, and the code starting at line 227 starts to execute,  before line 158, then the code starting in line 227 (which is holding `m_timer_lock`), will assign `m_update_sync_ctx = nullptr`. Then after line 158 we acquire the `m_timer_lock` but we will not call `add_event_after` anymore.